### PR TITLE
Fuse gemm add with pointwise fusions

### PR DESCRIPTION
**Pull request by pfultz2**

### CODE REVIEW

File: src/targets/gpu/fuse_ops.cpp [hunk 1]
```diff
@@ -931,6 +931,58 @@ struct find_gemm_add
...
+        auto gemm_ins = r.instructions["gemm"];
+        auto c_ins    = r.instructions["c"];
+
+        auto gemm = any_cast<rocblas_gemm<op::dot>>(gemm_ins->get_operator());
+
+        // Already fused gemm
+        if(not float_equal(gemm.beta, 0))
+            return;
+
+        if(std::any_of(ins->inputs().begin(), ins->inputs().end(), [](auto i) {
```
Comment (umangyadav) [inline src/targets/gpu/fuse_ops.cpp:970]: Do we need this ? We are checking for standard shapes of inputs in the matcher.

File: src/targets/gpu/fuse_ops.cpp [hunk 2]
```diff
@@ -931,6 +931,58 @@ struct find_gemm_add
...
+        auto n = std::count_if(pm->begin(), pm->end(), [&](auto& i) { return i.name() == s; });
+        if(n != 1)
+            return false;
+        return std::all_of(pm->begin(), pm->end(), [&](auto& i) {
+            return starts_with(i.name(), "@") or i.name() == s;
+        });
+    }));
+}
+
+struct find_gemm_pointwise
```
Comment (umangyadav) [inline src/targets/gpu/fuse_ops.cpp:947]: Do we have a test for this ?

Comment (pfultz2) [reply src/targets/gpu/fuse_ops.cpp:947]: We have verify tests for this.

File: src/targets/gpu/fuse_ops.cpp (hunk 1 above)
Comment (pfultz2) [reply src/targets/gpu/fuse_ops.cpp]: Removed.

Comment (umangyadav) [reply src/targets/gpu/fuse_ops.cpp]: It can be removed from `find_gemm_add` as well.

**Review by umangyadav — APPROVED** [review]

**Review by kahmed10 — CHANGES_REQUESTED** [review]
Can we remove L911-914 as well? Otherwise LGTM

**Review by kahmed10 — APPROVED** [review]

**Pull request merged**